File: src/main/java/training/katas/SplitStrings/split_strings.rb
def solution(str)
  if str.length == 0
    []
  end
  result = []
  (0..str.length - 1).step(2).each do |index|
    char1 = str[index]
    char2 = index + 1 < str.length ? str[index + 1] : '_'
    result.push(char1 + char2)
  end
  result
end

File: src/main/java/training/katas/SplitStrings/split_strings.py
def solution(s):
    print(s)
    if len(s) == 0:
        return []
    print([s[index] + (s[index + 1] if index + 1 < len(s) else '_')
           for index in range(0, len(s), 2)])
    return [s[index] + (s[index + 1] if index + 1 < len(s) else '_')
            for index in range(0, len(s), 2)]
